File: Sandtree/Game/Sandtree/Assets/PlayerMelee.js
#pragma strict

var movementScript	: PlayerMovement;
var camScript		: CamScript;
var animator		: Animator;

var attackVelocity	: float;

var canAttack	: boolean;
var attackTime	: float;
var attackInt	: int;

function Start ()
{
	movementScript	= GameObject.Find ("PlayerPrototype").GetComponent (PlayerMovement);
	camScript		= GameObject.Find ("CamObj").GetComponent (CamScript);
}

function Update ()
{
	if (Input.GetButtonDown ("Fire1"))
	{
		if (canAttack)
		{
			AttackFunc ();
		}
	}
}

function AttackFunc ()
{
//	Debug.Log ("AttackFunc");
	attackInt	= Random.Range (1,4);
	animator.SetInteger	("AttackInt", attackInt);
	VarAssign	();
	
	canAttack					= false;
	movementScript.inControl	= false;
	camScript.inControl			= false;
	
	var velYStore	= rigidbody.velocity.y;
	var clickVector	: Vector3;
	clickVector	= (camScript.pointVector - transform.position).normalized;
	rigidbody.velocity		= clickVector * attackVelocity;
	rigidbody.velocity.y	= velYStore;
	
	yield WaitForSeconds (attackTime * 0.5);
	//Put second attack stuff here
	
	yield WaitForSeconds (attackTime * 0.5);
	animator.SetInteger ("AttackInt", 0);
	canAttack					= true;
	movementScript.inControl	= true;
	camScript.inControl			= true;
}

function VarAssign ()
{
	switch (attackInt)
	{
		case (1):
			attackTime	= 0.66;
			attackVelocity	= 3;
			break;
		case (2):
			attackTime	= 0.8;
			attackVelocity	= 3;
		case (3):
			attackTime	= 0.75;
			attackVelocity	= 3;
	}
}

File: Sandtree/Game/Sandtree/Assets/Code/PlayerMelee.cs
using UnityEngine;
using System.Collections;

public class PlayerMelee : MonoBehaviour {

	public PlayerMovement movementScript;
	public CamScript camScript;
	public Animator animator;

	public float attackRange;
	public float attackVelocity;

	public Transform attackTarg;
	public bool canAttack;
	public float attackTime;
	public int attackInt;
	public float attackDis;
	public float reach;

	void Start ()
	{
		//	movementScript	= GameObject.Find ("PlayerPrototype").GetComponent (PlayerMovement);
		//	camScript		= GameObject.Find ("CamObj").GetComponent (CamScript);
	}
	
	void Update ()
	{
		if (Input.GetButtonDown ("Fire1"))
		{
			if (canAttack)
			{
				if (attackTarg != null)
				{	TargetedAttackFunc ();	}
				else
				{	UntargetedAttackFunc ();	}
			}
		}
	}
	
	IEnumerator TargetedAttackFunc ()
	{
		attackDis	= Vector3.Distance (attackTarg.position, transform.position);
		if (attackDis > attackRange)
		{
			UntargetedAttackFunc ();
		}
		else
		{
			//		Debug.Log ("TargetedAttack");
			Vector3 rigVelStore = rigidbody.velocity;
			rigVelStore.x = 0;
			rigVelStore.z = 0;
			rigidbody.velocity = rigVelStore;

			attackInt	= Random.Range (1,4);
			animator.SetInteger	("AttackInt", attackInt);
			VarAssign	();
			
			canAttack					= false;
			movementScript.inControl	= false;
			camScript.inControl			= false;
			
			float velYStore	= rigidbody.velocity.y;
			Vector3 clickVector;
			
			if (attackDis > reach)
			{
				attackVelocity	= (attackDis - reach) / (attackTime * 0.7f);
			}
			else if (attackDis < (reach + 0.25f))
			{
				attackVelocity	= -1;
			}
			
			clickVector	= (camScript.pointVector - transform.position).normalized;
			rigidbody.velocity		= clickVector * attackVelocity;
			Vector3 rigVelStore2 = rigidbody.velocity;
			rigVelStore2.y = velYStore;
			rigidbody.velocity = rigVelStore2;
			
			yield return new WaitForSeconds (attackTime * 0.5f);
			//Put second attack stuff here
			
			yield return new WaitForSeconds (attackTime * 0.5f);
			animator.SetInteger ("AttackInt", 0);
			canAttack					= true;
			movementScript.inControl	= true;
			camScript.inControl			= true;
			attackTarg			= null;
			camScript.hasTarget	= false;
		}
	}
	
	IEnumerator UntargetedAttackFunc ()
	{
		//	Debug.Log ("UntargetedAttack");
		attackInt	= Random.Range (1,4);
		animator.SetInteger	("AttackInt", attackInt);
		VarAssign	();
		
		canAttack					= false;
		movementScript.inControl	= false;
		camScript.inControl			= false;
		
		float velYStore	= rigidbody.velocity.y;
		Vector3 clickVector;
		attackVelocity	= 2;
		clickVector	= (camScript.pointVector - transform.position).normalized;
		rigidbody.velocity		= clickVector * attackVelocity;
		Vector3 rigVelStore3 = rigidbody.velocity;
		rigVelStore3.y = velYStore;
		rigidbody.velocity = rigVelStore3;

		yield return new WaitForSeconds (attackTime * 0.5f);
		//Put second attack stuff here
		
		yield return new WaitForSeconds (attackTime * 0.5f);
		animator.SetInteger ("AttackInt", 0);
		canAttack					= true;
		movementScript.inControl	= true;
		camScript.inControl			= true;
	}
	
	void VarAssign ()
	{
		switch (attackInt)
		{
		case (1):		//Front Stab
			attackTime	= 0.66f;
			reach		= 1f;
			break;
		case (2):		//Under Stab
			attackTime	= 0.7f;
			reach		= 1f;
			break;
		case (3):		//Spin Slash
			attackTime	= 0.75f;
			reach		= 0.9f;
			break;
		}
	}
}
